src/main.rs

```diff
@@ -251,8 +270,29 @@ fn run() -> Result<(), Box<::std::error::Error>> {
 
     let stdin = io::stdin();
 
+    let start_offset = if let Some(start) = matches
+        .value_of("start")
+        .and_then(|s| s.parse::<u64>().ok())
+    {
+        start
+    } else {
+        0
+    };
+    let mut print_offset = false;
+    let print_idx = start_offset - (start_offset % 16);
+
     let mut reader: Box<dyn Read> = match matches.value_of("file") {
-        Some(filename) => Box::new(File::open(filename)?),
+        Some(filename) => {
+            let mut f = File::open(filename)?;
+
+            // apply start offset
+            if start_offset > 0 {
+                f.seek(SeekFrom::Start(start_offset))
```

Comment:
So, question: is this really an error if, say, the file simply wasn't as big as the specified offset? I think that this is a good default behavior, but I'm not sure about the case where the file size simply doesn't match.

This applies to the `stdin` discussion above since it's entirely possible to detect if there are any remaining bytes in `stdin`.